### PR TITLE
Start on refactor for the code to amr pipeline to be able to handle...

### PR DESCRIPTION
**Pull request by Sorrento110**
dynamics selections in TDS and entire repos.

# Description

* Include a summary of the changes and the related issue. 
* Include relevant motivation and context.

Resolves #(issue)

### CODE REVIEW

File: api/server.py [hunk 1]
```diff
@@ -101,7 +101,11 @@ def equations_to_amr(
 
 @app.post("/code_to_amr")
 def code_to_amr(
-    code_id: str, name: Optional[str] = None, description: Optional[str] = None, redis=Depends(get_redis)
+    code_id: str,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    dynamics_only: Optional[bool] = False,
```
Comment (mattprintz) [inline api/server.py:107]: Can we add this param to the doc string(s)?

File: api/server.py [hunk 2]
```diff
@@ -195,7 +204,9 @@ def profile_dataset(
 
 
 @app.post("/profile_model/{model_id}")
-def profile_model(model_id: str, document_id: str, redis=Depends(get_redis)) -> ExtractionJob:
+def profile_model(
+    model_id: str, document_id: str, redis=Depends(get_redis)
+) -> ExtractionJob:
```
Comment (mattprintz) [inline api/server.py:209]: Function definitions should either be on a single line or have each param on a different line. Can we expand this out?

File: api/server.py [hunk 3]
```diff
@@ -217,8 +228,9 @@ def profile_model(model_id: str, document_id: str, redis=Depends(get_redis)) ->
 
 
 @app.post("/link_amr")
-def link_amr(document_id: str, model_id: str, redis=Depends(get_redis)) -> ExtractionJob:
-
+def link_amr(
+    document_id: str, model_id: str, redis=Depends(get_redis)
+) -> ExtractionJob:
```
Comment (mattprintz) [inline api/server.py:233]: Function definitions should either be on a single line or have each param on a different line. Can we expand this out?

File: tests/conftest.py [hunk 4]
```diff
@@ -129,8 +131,16 @@ def generate(code=False, **extra_params):
...
+            if dynamics_only:
+                with open(f"{context_dir}/dynamics.json", "r") as f:
+                    dynamics = json.load(f)
+            else:
+                dynamics = {}
+            code_file = {
+                "dynamics": dynamics,
+                "language": "python",
+            }
+            artifact["files"] = {"code.py": code_file}
```
Comment (mattprintz) [inline tests/conftest.py:143]: This is a pretty significant change in the structure of the artifact object.
It removes `filename` and `language` adds `files` which is a dict (compared to `file_names` which is a list) where the dict values now have 'dynamics' and 'language'. Feels kind of ripe for confusion/problems.

File: api/server.py (hunk 1 above)
Comment (Sorrento110) [reply api/server.py:107]: Yep let me do that!

File: tests/conftest.py (hunk 4 above)
Comment (brandomr) [reply tests/conftest.py:143]: @mattprintz yeah this is a major change--we'd discussed with @YohannParis and @tangopapatime but I still think has big ramifications and is one of the major reasons for our TA1 open house Friday

**Review by brandomr — APPROVED** [review]

**Pull request merged**